import subprocess
import traceback

from .database import DatabaseManagment
from .errors import Error
from .ToStdOut import ToStdout

error = Error
write = ToStdout.write

class ExploitHandler:
    def __init__(self, data):
        database = DatabaseManagment.get()
        try:
            write("Starting exploit")
            subprocess.run(["python3", database["EXPLOIT"]])
        except Exception:
            error(traceback.format_exc())